# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/konsole/login-shell.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- konsole-25.08.3/src/profile/Profile.cpp.vanilla	2025-12-07 16:50:37.225840126 +0200
+++ konsole-25.08.3/src/profile/Profile.cpp	2025-12-07 17:06:17.304000000 +0200
@@ -271,7 +271,7 @@
     setProperty(Path, BUILTIN_MAGIC_PATH);
     setProperty(Command, defaultShell());
     // See Pty.cpp on why Arguments is populated
-    setProperty(Arguments, QStringList{defaultShell()});
+    setProperty(Arguments, QStringList{defaultShell(), QString::fromUtf8("-ls")});
     setProperty(Font, QFontDatabase::systemFont(QFontDatabase::FixedFont));
 #if defined(Q_OS_WIN)
     setProperty(DefaultEncoding, QString::fromUtf8(QStringConverter::nameForEncoding(QStringConverter::Utf8)));
